src/algebra/char_p/two.lean

```diff
@@ -24,27 +24,27 @@ variables [semiring R] [char_p R 2]
 lemma two_eq_zero : (2 : R) = 0 :=
 by rw [← nat.cast_two, char_p.cast_eq_zero]
 
-lemma add_self_eq_zero (x : R) : x + x = 0 :=
+@[simp] lemma add_self_eq_zero (x : R) : x + x = 0 :=
 by rw [←two_smul R x, two_eq_zero, zero_smul]
 
-lemma bit0_eq_zero (x : R) : (bit0 x : R) = 0 :=
+@[simp] lemma bit0_eq_zero (x : R) : (bit0 x : R) = 0 :=
 add_self_eq_zero x
 
-lemma bit1_eq_one (x : R) : (bit1 x : R) = 1 :=
+@[simp] lemma bit1_eq_one (x : R) : (bit1 x : R) = 1 :=
```

Comment:
it may be worth making "unapplied" versions of these lemmas for `simp`

Reply:
I am not sure that I understand what these "unapplied" lemmas would be.  Do you mean moving the `(x : R)` to the other side of the colon?

Reply:
like `(bit1 : R → R) = 1` (if that compiles + doesn't mean the identity) and `(bit0 : R → R) = 0`

Reply:
I just pushed a version with the "unapplied" lemmas.  I removed the `simp` tag to the old ones, since they are now proved directly by `simp`!  Let me know if this is not what you had in mind!